.github/workflows/lint-and-test-matrix.yml

```diff
@@ -51,15 +51,22 @@ jobs:
         python -m pip install --upgrade pip
         pip install -e '.[all]'
 
-    - name: Run Tests with Coverage
+    - name: Run Tests
       run: |
-        pip install coverage pytest pytest-cov
-        coverage run -m pytest --maxfail=1 --disable-warnings -q
-        coverage report -m
-        coverage html
+        pip install pytest pytest-cov
+        pytest --cov --junitxml=junit.xml -o junit_family=legacy
 
-    - name: Upload coverage report
-      uses: actions/upload-artifact@v4
+    - name: Upload test results to Codecov
+      if: ${{ !cancelled() }}
+      uses: codecov/test-results-action@v1
       with:
-        name: coverage-html-report-${{ matrix.python-version }}
-        path: coverage_html_report
+        token: ${{ secrets.CODECOV_TOKEN }}
+
+    - name: Run Coverage
```

Comment:
Do we need this? Can we not run tests with coverage as a single action?

Reply:
I actually don't know. I will try to add the params to the first pytest call

Reply:
Looking at their example app it looks like they do 

```bash
pytest --cov app
```

See: https://github.com/codecov/example-python/blob/main/.github/workflows/ci.yml#L18

I guess in our case it'll be 

```bash
pytest --cov PyTado
```

Although not sure we need to specify the source dir as it's implicit with the tests :thinking: